### DISCUSSION TITLE
More kinds of printers wanted

### DISCUSSION THREAD

**Discussion by tongpeifu-yh** [86Box/86Box#2915] [Ideas]
I just want more kinds of printers to be emulated, especially which can be used under ms-dos, such as hp laser printer.   
Because one of the most popular programs that we Chinese old software enthusiasts like to explore are word processing systems, in which print is an important part.

**Comment by TC1995** [comment]
Documentation on the HP Laser printers? I mean, technical reference manuals regarding the commands, etc.

**Comment by tongpeifu-yh** [comment]
Do you mean pcl5 reference manual? Hp laserjet series use pcl as its print language. Here are the manuals that I download from hp official website.
[https://drive.google.com/drive/folders/1KMpSlax06M4NU5PP5jFWNxwwDjqXcIiy?usp=sharing](https://drive.google.com/drive/folders/1KMpSlax06M4NU5PP5jFWNxwwDjqXcIiy?usp=sharing)

**Comment by startmenu** [comment]
Ghostscript may convert PCL to PDF. Just capture LPT output and save as a file.

**Comment by taenologue** [comment]
Now that 86Box already has a generic PCL printer, I think what’s left is just to make it behave a bit closer to the real thing. Things like soft font download, macros/forms, and CJK font cartridge support were an important part of how LaserJets worked with DOS apps. Also, the old DOS utilities that upload fonts should ideally work with the emulated printer. If these parts get added later on, the emulation would feel much more complete.

**Comment by Hockey3858** [comment]
When they got a Generic PCL Printer

**Comment by taenologue** [comment]
Enable developer build by updating CMakeLists.txt.
 - 86Box-master/CMakeLists.txt
`option(DEV_BRANCH   "Development branch"                                         OFF) -> ON
`
